doc/cjdns.md

```diff
@@ -0,0 +1,92 @@
+# CJDNS support in Bitcoin Core
+
+It is possible to run Bitcoin Core over CJDNS, an encrypted IPv6 network that
+uses public-key cryptography for address allocation and a distributed hash table
+for routing.
+
+## What is CJDNS?
+
+CJDNS is like a distributed, shared VPN with multiple entry points where every
+participant can reach any other participant. All participants use addresses from
+the `fc00::/8` network (reserved IPv6 range). Installation and configuration is
+done outside of Bitcoin Core, similarly to a VPN (either in the host/OS or on
+the network router).
+
+Compared to IPv4/IPv6, CJDNS provides end-to-end encryption and protects nodes
+from traffic analysis and filtering.
+
+Used with Tor and I2P, CJDNS is a complementary option that can enhance network
+redundancy and robustness for both the Bitcoin network and individual nodes.
+
+Each network has different characteristics. For instance, Tor is widely used but
+somewhat centralized. I2P connections have a source address and I2P is slow.
+CJDNS is fast but does not hide the sender and the recipient from intermediate
+routers.
+
+## Installing CJDNS and connecting to the network
+
+To install and set up CJDNS, follow the instructions at
+https://github.com/cjdelisle/cjdns#cjdns.
+
+Don't skip steps
+["2. Find a friend"](https://github.com/cjdelisle/cjdns#2-find-a-friend) and
+["3. Connect your node to your friend's
+node"](https://github.com/cjdelisle/cjdns#3-connect-your-node-to-your-friends-node).
+You need to be connected to the CJDNS network before it will work with your
+Bitcoin Core node.
+
+Typically, CJDNS might be launched with `sudo ./cjdroute < cjdroute.conf` and it
+sheds permissions after setting up the [TUN](https://en.wikipedia.org/wiki/TUN/TAP)
+device. You may also [launch it as an unprivileged
+user](https://github.com/cjdelisle/cjdns/blob/master/doc/non-root-user.md) with
+some additional setup.
+
+The network connection can be checked with `./tools/peerStats`.
+
+## Run Bitcoin Core with CJDNS
+
+Once you are connected to the CJDNS network, the following Bitcoin Core
+configuration option makes CJDNS peers automatically reachable:
+
+```
+-cjdnsreachable
+```
+
+With this option enabled, if your node connects to `fc00::/8` addresses, they
+will lead to the CJDNS network. Without the option and by default, they will
+lead to an IPv6 local network as defined in
+[RFC 4193](https://datatracker.ietf.org/doc/html/rfc4193).
```

Comment:
This gives the wrong impression that Bitcoin Core will decide to connect to CJDNS instead of the private IPv6 network based on `-cjdnsreachable`. That is not the case. This option merely tells Bitcoin Core that it is running in such an environment that connecting to a `fc00::/8` address will result in connecting to the CJDNS network, not the IPv6 private network. Given a peer like `[fc12::34]:8333` Bitcoin Core cannot choose between CJDNS vs IPv6-private. I am not sure how to make this more explicit in the text above.

The `-cjdnsreachable` option helps Bitcoin Core do better address management. Like, if one of its local addresses is `fc00::/8` and `-cjdnsreachable` is set, then it may choose to gossip that address to others. Also, making it possible to distinguish whether an incoming connection from `fc00::/8` comes from the CJDNS network or from a IPv6-private one.

Reply:
Great feedback. I'll try to improve it.